src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -667,6 +670,14 @@ protected override bool TryParseRequest(ReadResult result, out bool endConnectio
             }
             throw;
         }
+#pragma warning disable CS0618 // Type or member is obsolete
+        catch (BadHttpRequestException ex)
+        {
+            DetectHttp2Preface(result.Buffer, ex);
+
+            throw;
```

Comment:
If the exception is re-thrown, isn't it going to try to write a 400 response after the GOAWAY?

Reply:
Calling CancelRequestAbortedToken sets _connectionAborted to true, which stops the 400 response. Instead, Output.FlushAsync is called (see comment on TryProduceInvalidRequestResponse change)